services/subscriptions/subscriptions-supabase.service.ts

```diff
@@ -20,16 +25,79 @@ class SubscriptionSupabase implements SubscriptionService {
   /**
    * retruns the subscription link, so that the users can subscribe to a plan
    */
-  public async getSubscriptionLink(): Promise<{ url: string | null }> {
-    const { data, error } = await this.supabase.functions.invoke(
-      "stripe-create-subscription"
-    );
-    if (error && error instanceof FunctionsHttpError) {
-      const errorResponse = await error.context.json();
-      messageHandler.handleError(errorResponse.message);
-      return { url: null };
+  public async getSubscriptionLink(): Promise<{ url: string | undefined }> {
+    try {
+      const { data, error } = await this.supabase.functions.invoke<{
+        checkoutUrl: string | undefined;
+      }>("stripe-create-subscription");
+      if (!error && data?.checkoutUrl) {
+        return { url: data?.checkoutUrl }
+      }
+      if (error instanceof FunctionsHttpError) {
+        const errorMessage = await error.context.json();
+        throw errorMessage?.message;
+      } else if (error instanceof FunctionsRelayError) {
+        throw error?.message;
+      } else if (error instanceof FunctionsFetchError) {
+        throw error?.message;
```

Comment:
We dont need to throw the error, we can handle the message here and return `null` or `{url:''}`

Reply:
Here I am throwing the error to catch the error message only, it should not go out side the try catch block, but now I have moved this part to the message-handler to have a common function for Edge function errors.